api/filters/replacement/replacement.go

```diff
@@ -247,3 +267,32 @@ func setFieldValue(options *types.FieldOptions, targetField *yaml.RNode, value *
 
 	return nil
 }
+
+func getByDelimiter(delimiter string, target string, source string, index int) string {
+	tv := strings.Split(target, delimiter)
+	// TODO: Add a way to remove an element
+	switch {
+	case index < 0: // prefix
+		tv = append([]string{source}, tv...)
+	case index >= len(tv): // suffix
+		tv = append(tv, source)
+	default: // replace an element
+		tv[index] = source
+	}
+	return strings.Join(tv, delimiter)
+}
+
+func getByRegex(regex string, target string, source string, index int) string {
```

Comment:
Since both these are fragile string manipulation methods, can we have separate tests to verify these and ensure we are capturing the edge cases?

Reply:
@varshaprasad96 I've written some edge case tests but if you want I can add some tests that handle these functions in specific